code/modules/paperwork/photography.dm

```diff
@@ -570,26 +610,69 @@
 	pictures_left--
 	to_chat(user, "<span class='notice'>[pictures_left] photos left.</span>")
 	icon_state = icon_off
-	on = 0
+	on = FALSE
 	if(pictures_left > 0)
 		spawn(64)
 			icon_state = icon_on
-			on = 1
+			on = TRUE
 
 /obj/item/device/camera/remote_attack(atom/target, mob/user, atom/movable/eye)
 	if(istype(eye, /obj/machinery/camera))
 		return afterattack(target, user) //Allow taking photos when looking through cameras
 
-/obj/item/device/camera/ai_camera/proc/toggle_camera_mode()
-	if(in_camera_mode)
-		camera_mode_off()
-	else
-		camera_mode_on()
+/obj/item/device/camera/silicon/proc/toggle_camera_mode(var/mob/living/silicon/S = null)
+	if(!S)
+		return
+	in_camera_mode = !in_camera_mode
+	to_chat(S, "<B>Camera Mode [in_camera_mode ? "activated":"deactivated"]</B>")
+	if(S.camera_icon)
+		S.camera_icon.icon_state = "camera[in_camera_mode ? "1":""]"
 
-/obj/item/device/camera/ai_camera/proc/camera_mode_off()
-	src.in_camera_mode = 0
-	to_chat(usr, "<B>Camera Mode deactivated</B>")
+/obj/item/device/camera/silicon/robot_camera/proc/borgprint()
+	var/list/nametemp = list()
+	var/find
+	var/datum/picture/selection
+	var/mob/living/silicon/robot/C = loc
+	var/obj/item/device/camera/silicon/targetcam = null
 
-/obj/item/device/camera/ai_camera/proc/camera_mode_on()
-	src.in_camera_mode = 1
-	to_chat(usr, "<B>Camera Mode activated</B>")
+	if(C.stat)
+		return
+	if(C.toner < CYBORG_PHOTO_COST)
+		to_chat(C, "Insufficent toner to print image.")
+		return
+	if(C.connected_ai)
+		targetcam = C.connected_ai.aicamera
+	else
+		targetcam = C.aicamera
+	if(!targetcam.aipictures.len)
+		to_chat(C, "<span class='danger'>No images saved</span>")
+		return
+	for(var/datum/picture/t in targetcam.aipictures)
+		nametemp += t.fields["name"]
+	find = input("Select image (numbered in order taken)") in nametemp
+	for(var/datum/picture/q in targetcam.aipictures)
+		if(q.fields["name"] == find)
+			selection = q
+			break
+	var/obj/item/weapon/photo/p = new /obj/item/weapon/photo(C.loc)
+	p.photocreate(selection.fields["icon"], selection.fields["img"], selection.fields["info"], selection.fields["blueprints"])
+	p.pixel_x = rand(-10, 10)
+	p.pixel_y = rand(-10, 10)
+	C.toner -= CYBORG_PHOTO_COST
+	visible_message("[C.name] spits out a photograph from a narrow slot on it's chassis.")
+	playsound(loc, "polaroid", 75, 1, -3)
+	to_chat(C, "You print a photograph.")
+
+/obj/item/device/camera/silicon/proc/sync(var/mob/living/silicon/robot/R)
+	if(R.connected_ai && R.connected_ai.aicamera && R.aicamera) // Send images the Cyborg has taken to the AI's album upon sync.
+		if(R.aicamera.aipictures.len)
+			for(var/i in R.aicamera.aipictures)
+				if(!i in R.connected_ai.aicamera.aipictures)
```

Comment:
Ackshually, can't you just rewrite the entirety of this into `R.connected_ai.aicamera.aipictures |= R.aicamera.aipictures`?

Reply:
You're right, this is dumb.